packages/ai/src/Button.ts

```diff
@@ -58,13 +87,185 @@ class Button extends UI5Element {
+	 * @public
+	 */
+	@slot({ type: ButtonState, "default": true })
+	states!: Array<ButtonState>;
+
+	onBeforeRendering(): void {
+		if (this.fadeOut || this.fadeIn) {
+			return;
+		}
+
+		if (this.state && this._currentStateObject && !Object.keys(this._currentStateObject).length) {
+			this._currentStateObject = this._findStateByName(this.state);
+		}
+
+		const currentStateName = this._currentStateObject?.name || "";
+
+		if (!this.state) {
+			this.state = this.states.length ? this.states[0].name : "";
+			this._currentStateObject = this._findStateByName(this.state);
+		}
+
+		if (!this._currentStateObject) {
+			this._throwMissingStateError();
+		}
+
+		if (currentStateName !== "" && currentStateName !== this.state) {
+			this._fadeOut();
+		}
+	}
+
+	onAfterRendering(): void {
+		if (!this._findStateByName(this.state!) && this._currentStateObject) {
+			this.state = this._currentStateObject.name;
+		}
+	}
+
+	/**
+	 * Starts the fade-out animation.
+	 * @private
+	 */
+	async _fadeOut(): Promise<void> {
+		const button = this.shadowRoot?.querySelector("[ui5-button]") as MainButton;
+		const newStateObject = this._findStateByName(this.state!);
+		if (button && newStateObject) {
+			const buttonWidth = (button as HTMLElement).offsetWidth;
+			const hiddenButton = this.shadowRoot?.querySelector(".ui5-ai-button-hidden") as MainButton;
+			button.style.width = `${buttonWidth}px`;
+			hiddenButton.icon = newStateObject.icon;
+			hiddenButton.endIcon = newStateObject.endIcon;
+			hiddenButton.textContent = newStateObject.text;
+
+			await renderFinished();
+			const hiddenButtonWidth = (hiddenButton as HTMLElement).offsetWidth;
+			this.fadeOut = true;
+			button.style.width = `${hiddenButtonWidth}px`;
+
+			setTimeout(() => {
+				this.fadeMid = true;
+				this._currentStateObject = newStateObject;
+				this._fadeIn();
+			}, AI_BUTTON_FADE_OUT_DURATION);
+		} else {
+			this._throwMissingStateError();
+		}
+	}
+
+	/**
+	 * Starts the fade-in animation.
+	 * @private
+	 */
+	_fadeIn(): void {
+		setTimeout(() => {
+			this.fadeIn = true;
+			this._resetFade();
+		}, AI_BUTTON_FADE_IN_DURATION);
+	}
+
+	/**
+	 * Resets the fade phases when the animation is completed.
+	 * @private
+	 */
+	_resetFade(): void {
+		setTimeout(() => {
+			this.fadeOut = false;
+			this.fadeMid = false;
+			this.fadeIn = false;
+		}, AI_BUTTON_FADE_RESET_DURATION);
+	}
+
+	/**
+	 * Returns the current state object.
+	 * @private
+	 */
+	_findStateByName(name: string): ButtonState | undefined {
+		return this.states.find(state => state.name === name);
+	}
+
+	/**
+	 * Handles the click event.
+	 * @private
+	 */
+	_onclick(e: MouseEvent): void {
+		e.stopImmediatePropagation();
+		this.fireEvent("click");
+	}
+
+	/**
+	 * Throws an error when the current state is missing.
+	 * @private
+	 */
+	_throwMissingStateError(): void {
+		throw new Error(`State with name="${this.state}" doesn't exist!`);
```

Comment:
I wonder if we need to throw an error, or we can just set the current state as next state and log an error in the console. As throwing an error will interrupt the function. That's up for discussion

Reply:
We cannot set next state as we don't know the current use case - maybe previous will be better in some cases, and in other cases some third will be fine - all depends on app logic here. And YES, I think the function should be interrupted if there is "wrong/missing" state set, because otherwise something void can happen.